### ISSUE TITLE
tsc --watch does not correctly account for added & deleted files

### ISSUE THREAD

**Issue by dgoldstein0** [Microsoft/TypeScript#27585]
**TypeScript Version:**  3.1.1

**Search Terms:** tsc --watch

**Code**

```ts
// a.ts
declare module 'a' {
  type foo = number;
}

// tsconfig.json
{}
```

Run `tsc --watch --noEmit --project .`.  Wait for typechecking to complete.  `cp a.ts b.ts`; wait for typechecking to complete again.  then `touch *` and wait for typechecking to run again.  then `rm b.ts` and wait for typechecking to complete again

**Expected behavior:**

Typechecking should succeed on the first run, should fail with 2 errors on the second run (`error TS2300: Duplicate identifier 'foo'`) since a.ts and b.ts duplicate the same module; the third run should come up with the same results, and the 4th run should succeed.

**Actual behavior:**

The first run passes as expected, the second run only detects the error in `b.ts`, the touch * causes the 3rd run to recognize both errors, and then deleting b.ts causes the 4th run to still think there's an error in `a.ts`.

If I had to guess, tsc --watch is assuming in these file add / remove cases that adding or removing b.ts isn't going to change the type errors it finds in a.ts; this type of error is an example where that's clearly not the case.

**Related Issues:** couldn't find any

**Comment by sheetalkamat** [comment]
I don't think this is by design. Its a bug as in we do not re-retrieve semantic diagnostics of the file which could share the new added augmented module. We need to add that logic to builder/builderState.

**Comment by dgoldstein0** [comment]
thanks for the quick turnaround!  I'll look forward to the fix in TS 3.2  And in the meantime if I find more bugs of this type I'll report them too.
